discord/app_commands/models.py

```diff
@@ -195,6 +206,127 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f'<{self.__class__.__name__} id={self.id!r} name={self.name!r} type={self.type!r}>'
 
+    @property
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`~discord.Guild`]: Returns the guild this command is registered to
+        if it exists.
+        """
+        if self.guild_id and self._state:
+            return self._state._get_guild(self.guild_id)
+
+        return None
+
+    async def delete(self) -> None:
+        """|coro|
+
+        Deletes the application command.
+
+        Raises
+        -------
+        NotFound
+            The application command was not found.
+        Forbidden
+            You do not have permission to delete this application command.
+        HTTPException
+            Deleting the application command failed.
+        ClientException
+            The client does not have an application ID.
+        """
+        state = self._state
+        if not state.application_id:
+            raise ClientException(APP_ID_NOT_FOUND)
+
+        if self.guild_id:
+            await state.http.delete_guild_command(
+                state.application_id,
+                self.guild_id,
+                self.id,
+            )
+        else:
+            await state.http.delete_global_command(
+                state.application_id,
+                self.id,
+            )
+
+    async def edit(
+        self,
+        *,
+        name: str = MISSING,
+        description: str = MISSING,
+        default_member_permissions: Optional[Permissions] = MISSING,
+        dm_permission: bool = MISSING,
+    ) -> AppCommand:
```

Comment:
I feel like there should be a way to edit the options of a command but right now those types don't seem user constructible.